lib/model/basket.rb

```diff
@@ -26,14 +26,18 @@ def total_price_gross
 
     private
     
-    def set_id
+    def build_id
       @@id += 1
     end
 
-    def set_quantity(quantity)
+    def check_quantity(quantity)
       raise ArgumentError unless quantity.is_a?(Numeric)
       raise ArgumentError if quantity <= 0
       quantity
     end
+ 
+    def basket_sum_gross
+      basket_sum_gross = FetchBasketItems.new.call.map { |basket_item|  basket_item[:total_price_gross] }.reduce(:+)
```

Comment:
this line look too long. btw check if sum is correct in the case of an empty basket.